### PR TITLE
Update copyright notice

### LINKED ISSUE #454: Update copyright notice
At least the primary LICENSE file should be updated from
> Copyright (c) 2014 HashiCorp, Inc.

to a generic attribution/copyright holder, for example:

> Copyright (c) 2014-2023 HashiCorp, Inc.
> Copyright (c) 2023- OpenTofu contributors

See also https://reuse.software/spec/#format-of-copyright-notices

To avoid any confusion it could also be useful to explicitly state that contributors are not assigning or transferring any copyrights to HashiCorp.

### PR DESCRIPTION
**Pull request by namgyalangmo**
Workflow:
1) Run "copywrite headers" in the terminal or run go generate to add headers to files without copyrights.
2) Run "go generate ./..." to do the same.
Feel free to run "copywrite --help" for more information about the tool
Note:
1)Added exclusion for *.proto files in .copywrite.hcl  to prevent a race between protobuf generation and copyright generation.
2)The process of invoking copyright remains unchanged despite modifications made in the .copywrite.hcl file. Initially, the .copywrite.hcl file assigned copyrights to HashiCorp by default. However, I have now updated the copyright holder by assigning it to the copyright_holder variable within the same file, effectively updating the copyright information.

Resolves #454 

## Target Release

1.7.0

### CODE REVIEW

**Comment by cube2222** [comment]
Sorry, that was my bad - we should only be ignoring .pb.go files (as those are generated), .proto files should not be ignored.

Happy to do that as a post-merge fix, though.

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,7 @@
-Copyright (c) 2014 HashiCorp, Inc.
-
-Mozilla Public License, version 2.0
```
Comment (cube2222) [inline LICENSE:3]: This file needs to be skipped as well.
Instead, the header should be manually changed to:
```
Copyright (c) The OpenTofu Authors
Copyright (c) 2014 HashiCorp, Inc.

Mozilla Public License, version 2.0
```

Comment (namgyalangmo) [reply LICENSE:3]: Will do

Comment (namgyalangmo) [reply LICENSE:3]: Just pushed the requested changes, i.e removed the *.proto file from ignore list and manually updated the LICENSE file. We don't need to add the license file to the header ignore list as copywrite tool doesn't update the preexisting copyrights, it just adds copyrights to the files that are missing copyright entirely (verified by running copywrite headers --plan).

**Review by cube2222 — APPROVED** [review]

**Comment by Yantrio** [comment]
Whilst reviewing this PR, I removed a copyright notice from a random go file. Then ran `make generate` which runs `go generate ./...` under the hood.

Am I correct in assuming that it should add the copyright header back in? Because it did not.

**Comment by Yantrio** [comment]
Update:

It looks like because there's the `tools` build tag in the file.

We will need to call either `go generate ./copyright_headers.go` or `go run github.com/hashicorp/copywrite headers` manually

**Comment by namgyalangmo** [comment]
> Update:
> 
> It looks like because there's the `tools` build tag in the file.
> 
> We will need to call either `go generate ./copyright_headers.go` or `go run github.com/hashicorp/copywrite headers` manually
> 
> I'll approve, but we need to look into this more in the future. And make a choice if we even want to carry on using this copywrite tool or not.
> 
> After running this command manually, I can see that another large set of files have their copyright updated. @namgyalangmo would you be able to discuss why these changes are not made in your PR?
> 
> What changed for me:
> 
> ```
> On branch update-copyright-notice
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   .github/ISSUE_TEMPLATE/rfc.yml
>         modified:   .goreleaser.yaml
>         modified:   cmd/tofu/telemetry.go
>         modified:   internal/addrs/check_rule_diagnostic.go
>         modified:   internal/addrs/check_rule_diagnostic_test.go
>         modified:   internal/backend/remote-state/consul/Dockerfile
>         modified:   internal/backend/remote-state/s3/backend_complete_test.go
>         modified:   internal/backend/remote-state/s3/diags.go
>         modified:   internal/backend/remote-state/s3/testing_test.go
>         modified:   internal/backend/remote-state/s3/validate.go
>         modified:   internal/backend/remote-state/s3/validate_test.go
>         modified:   internal/cloudplugin/cloudplugin1/grpc_client.go
>         modified:   internal/cloudplugin/cloudplugin1/grpc_client_test.go
>         modified:   internal/cloudplugin/cloudplugin1/grpc_plugin.go
>         modified:   internal/cloudplugin/cloudproto1/cloudproto1.proto
>         modified:   internal/cloudplugin/interface.go
>         modified:   internal/command/alias.go
>         modified:   internal/command/arguments/test.go
>         modified:   internal/command/arguments/test_test.go
>         modified:   internal/command/cliconfig/util.go
>         modified:   internal/command/cloud.go
>         modified:   internal/command/e2etest/test_test.go
>         modified:   internal/command/init_test.go
>         modified:   internal/command/telemetry.go
>         modified:   internal/command/test.go
>         modified:   internal/command/test_test.go
>         modified:   internal/command/testing/test_provider.go
>         modified:   internal/command/views/json/test.go
>         modified:   internal/command/views/test.go
>         modified:   internal/command/views/test_test.go
>         modified:   internal/configs/configschema/filter.go
>         modified:   internal/configs/configschema/filter_test.go
>         modified:   internal/configs/test_file.go
>         modified:   internal/configs/test_file_test.go
>         modified:   internal/genconfig/doc.go
>         modified:   internal/genconfig/generate_config.go
>         modified:   internal/genconfig/generate_config_test.go
>         modified:   internal/genconfig/generate_config_write.go
>         modified:   internal/lang/funcs/refinements.go
>         modified:   internal/moduletest/file.go
>         modified:   internal/moduletest/run.go
>         modified:   internal/moduletest/run_test.go
>         modified:   internal/moduletest/status.go
>         modified:   internal/moduletest/suite.go
>         modified:   internal/plans/planfile/wrapped.go
>         modified:   internal/tfdiags/override.go
>         modified:   internal/tfdiags/override_test.go
>         modified:   internal/tofu/eval_import.go
>         modified:   internal/tofu/eval_import_test.go
>         modified:   internal/tofu/node_external_reference.go
>         modified:   internal/tofu/test_context.go
>         modified:   internal/tofu/test_context_test.go
>         modified:   internal/tofu/transform_check_starter.go
>         modified:   internal/tofu/transform_external_reference.go
>         modified:   internal/tofumigrate/tofumigrate.go
>         modified:   internal/tofumigrate/tofumigrate_test.go
>         modified:   testing/equivalence-tests/tests/basic_json_string_update/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/basic_multiline_string_update/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/drift_refresh_only/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/drift_relevant_attributes/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/drift_simple/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/fully_populated_complex_destroy/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/fully_populated_complex_update/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/local_provider_delete/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/local_provider_update/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/moved_simple/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/moved_with_drift/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/moved_with_refresh_only/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/moved_with_update/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/null_provider_delete/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/replace_within_list/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/replace_within_map/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/replace_within_object/.terraform.lock.hcl
>         modified:   testing/equivalence-tests/tests/replace_within_set/.terraform.lock.hcl
>         modified:   website/.eslintrc.js
>         modified:   website/Dockerfile
>         modified:   website/docker-compose.build.yml
>         modified:   website/docker-compose.yml
>         modified:   website/docs/cli/commands/test/examples/expect_failures_resources/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/expect_failures_variables/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/module/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/nested-layout/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/nested-layout/module/tests/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/offline/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/plan/Dockerfile
>         modified:   website/docs/cli/commands/test/examples/plan/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/provider_alias/Dockerfile
>         modified:   website/docs/cli/commands/test/examples/provider_alias/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/simple/main.tftest.hcl
>         modified:   website/docs/cli/commands/test/examples/test.sh
>         modified:   website/docs/cli/commands/test/examples/variables/main.tftest.hcl
>         modified:   website/docs/intro/install/alpine-convenience.sh
>         modified:   website/docs/intro/install/alpine-manual.sh
>         modified:   website/docs/intro/install/alpine.sh
>         modified:   website/docs/intro/install/brew-install.sh
>         modified:   website/docs/intro/install/brew.sh
>         modified:   website/docs/intro/install/buildkite.jsx
>         modified:   website/docs/intro/install/deb-convenience.sh
>         modified:   website/docs/intro/install/deb-step1.sh
>         modified:   website/docs/intro/install/deb-step2.sh
>         modified:   website/docs/intro/install/deb-step3.sh
>         modified:   website/docs/intro/install/deb-step4.sh
>         modified:   website/docs/intro/install/deb.sh
>         modified:   website/docs/intro/install/docker-compose.yaml
>         modified:   website/docs/intro/install/install-opentofu.sh
>         modified:   website/docs/intro/install/install-yum.sh
>         modified:   website/docs/intro/install/install-zypper.sh
>         modified:   website/docs/intro/install/repo-yum.sh
>         modified:   website/docs/intro/install/repo-zypper.sh
>         modified:   website/docs/intro/install/rpm-convenience.sh
>         modified:   website/docs/intro/install/rpm.sh
>         modified:   website/docs/intro/install/standalone-install.ps1
>         modified:   website/docs/intro/install/standalone-install.sh
>         modified:   website/docs/intro/install/test-install-instructions.sh
>         modified:   website/docs/intro/install/verify-checksum-linux.sh
>         modified:   website/docs/intro/install/verify-checksum-macos.sh
>         modified:   website/docs/intro/install/verify-checksum.ps1
>         modified:   website/docs/intro/install/verify-cosign.ps1
>         modified:   website/docs/intro/install/verify-cosign.sh
> ```

@Yantrio  I actually brought that up here [https://github.com/opentofu/opentofu/issues/454](url) , The files mentioned were devoid of copyright notices when I forked the repo. I left them as is, wondering if there was a specific reason for their omission. Additionally, I'm puzzled as to why these files lacked a copyright notice, especially considering the tool's recursive nature and the absence of these files in the header_ignore list.

**Comment by Yantrio** [comment]
I believe we should add the copyright notice to all files. Including files that we made ourselves, and those without it.

After speaking to @cube2222 about this, we think that the hashicorp AND opentofu license notices should be added to all files that did not have a license before too.

Please add the license to those files too.

**Comment by namgyalangmo** [comment]
> I believe we should add the copyright notice to all files. Including files that we made ourselves, and those without it.
> 
> After speaking to @cube2222 about this, we think that the hashicorp AND opentofu license notices should be added to all files that did not have a license before too.
> 
> Please add the license to those files too.

On it!

**Comment by namgyalangmo** [comment]
Last time the the code consistency check threw this error,  
proto error
as mentioned on line no. 28 , I ran make generate and make protobuf. It updated a .pg.go file,
Screenshot 2024-02-07 at 9 17 00 PM
 @cube2222  should I go ahead and push the changes as instructed or will adding the copyright generate inconsistent results and instead I should remove the copyright form cloudproto1.proto file?

**Comment by Yantrio** [comment]
Let's keep the header in, please commit this pb.go files. Them having the header is correct. It looks like the protobuf tooling does copy those across.

**Comment by namgyalangmo** [comment]
> Let's keep the header in, please commit this pb.go files. Them having the header is correct. It looks like the protobuf tooling does copy those across.

done.

**Comment by Yantrio** [comment]
Thank you for this contribution @namgyalangmo 👍 Everything is looking good and I'll merge.

**Pull request merged**